source/isaaclab/isaaclab/sim/utils.py

```diff
@@ -570,7 +573,72 @@ def make_uninstanceable(prim_path: str | Sdf.Path, stage: Usd.Stage | None = Non
             # make the prim uninstanceable
             child_prim.SetInstanceable(False)
         # add children to list
-        all_prims += child_prim.GetChildren()
+        all_prims += child_prim.GetFilteredChildren(Usd.TraverseInstanceProxies())
+
+
+def resolve_world_scale(prim: Usd.Prim) -> tuple[float, float, float]:
+    """Resolve the world scale of a prim.
+
+    Args:
+        prim: The USD prim to resolve the world scale for.
+
+    Returns:
+        The world scale of the prim in the x, y, and z directions.
+    """
+
+    xform = UsdGeom.Xformable(prim)
+    world_transform = xform.ComputeLocalToWorldTransform(Usd.TimeCode.Default())
+    return tuple([*(v.GetLength() for v in world_transform.ExtractRotationMatrix())])
+
+
+def resolve_world_pose(prim: Usd.Prim) -> tuple[list[float], list[float]]:
+    """Resolve the world pose (position and orientation) of a prim.
+
+    Args:
+        prim: The USD prim to resolve the world pose for.
+
+    Returns:
+        A tuple containing the world position (as a 3D vector) and the world orientation (as a quaternion, format wxyz).
+    """
+
+    xform = UsdGeom.Xformable(prim)
+    world_transform = xform.ComputeLocalToWorldTransform(Usd.TimeCode.Default())
+    world_position = [*world_transform.ExtractTranslation()]
+    world_orientation = [
+        world_transform.ExtractRotation().GetQuaternion().GetNormalized().real,
+        *world_transform.ExtractRotation().GetQuaternion().GetNormalized().imaginary,
+    ]
+    return world_position, world_orientation
+
+
+def resolve_relative_pose(
+    prim: Usd.Prim, relative_to: Usd.Prim | None = None, device: torch.device | None = None
+) -> tuple[torch.Tensor, torch.Tensor]:
+    """Resolve the relative pose (position and orientation) of a prim with respect to another prim.
+
+    Args:
+        prim: The USD prim to resolve the relative pose for.
+        relative_to: The USD prim to resolve the relative pose with respect to. If None, the world frame is used.
+            Defaults to None.
+        device: The torch device to use for tensor operations. If None, the default device is used.
+            Defaults to None.
+
+    Returns:
+        A tuple containing the relative position (as a 3D vector) and the relative orientation (as a quaternion, format wxyz).
+    """
+
+    prim_pos, prim_quat = resolve_world_pose(prim)
+    if relative_to is None:
+        return torch.tensor(prim_pos, device=device), torch.tensor(prim_quat, device=device)
+
+    relative_to_pos, relative_to_quat = resolve_world_pose(relative_to)
```

Comment:
Device isn't resolved correctly.

Reply:
Instead of torch operation, we can also simply use pxr GfMat4 math

Reply:
Merging in your PR instead of these changes